src/cloud-resources/src/crd/console.rs

```diff
@@ -0,0 +1,145 @@
+        pub console_image_ref: String,
+        // Resource requirements for the console pod
+        pub resource_requirements: Option<ResourceRequirements>,
+        // Number of console pods to create
+        pub replicas: Option<i32>,
+        // The configuration for generating an x509 certificate using cert-manager for console
+        // to present to incoming connections.
+        // The dns_names and issuer_ref fields are required.
+        pub external_certificate_spec: Option<MaterializeCertSpec>,
+        // Annotations to apply to the pods
+        pub pod_annotations: Option<BTreeMap<String, String>>,
+        // Labels to apply to the pods
+        pub pod_labels: Option<BTreeMap<String, String>>,
+
+        // Connection information for the balancerd service to use
+        pub balancerd: BalancerdRef,
+
+        // This can be set to override the randomly chosen resource id
+        pub resource_id: Option<String>,
+    }
+
+    impl Console {
+        pub fn name_prefixed(&self, suffix: &str) -> String {
+            format!("mz{}-{}", self.resource_id(), suffix)
+        }
+
+        pub fn resource_id(&self) -> &str {
+            &self.status.as_ref().unwrap().resource_id
+        }
+
+        pub fn deployment_name(&self) -> String {
+            self.name_prefixed("console")
+        }
+
+        pub fn replicas(&self) -> i32 {
+            self.spec.replicas.unwrap_or(2)
+        }
+
+        pub fn app_name(&self) -> String {
+            "console".to_owned()
+        }
+
+        pub fn service_name(&self) -> String {
+            self.name_prefixed("console")
+        }
+
+        pub fn configmap_name(&self) -> String {
+            self.name_prefixed("console")
+        }
+
+        pub fn external_certificate_name(&self) -> String {
+            self.name_prefixed("console-external")
+        }
+
+        pub fn external_certificate_secret_name(&self) -> String {
+            self.name_prefixed("console-external-tls")
+        }
+
+        pub fn status(&self) -> ConsoleStatus {
+            self.status.clone().unwrap_or_else(|| ConsoleStatus {
+                resource_id: self
+                    .spec
+                    .resource_id
+                    .clone()
+                    .unwrap_or_else(new_resource_id),
+                conditions: vec![],
+            })
+        }
+    }
+
+    #[derive(Clone, Debug, Default, Deserialize, Serialize, JsonSchema, PartialEq)]
+    #[serde(rename_all = "camelCase")]
+    pub struct ConsoleStatus {
+        /// Resource identifier used as a name prefix to avoid pod name collisions.
+        pub resource_id: String,
+
+        pub conditions: Vec<Condition>,
+    }
+
+    impl ManagedResource for Console {
+        fn default_labels(&self) -> BTreeMap<String, String> {
+            BTreeMap::from_iter([(
+                "materialize.cloud/mz-resource-id".to_owned(),
+                self.resource_id().to_owned(),
+            )])
+        }
```

Comment:
Should we also have a label saying it is the console? Maybe `materialize.cloud/app: console`?